soci/soci_index.go

```diff
@@ -405,8 +428,11 @@ func (b *IndexBuilder) buildSociLayer(ctx context.Context, desc ocispec.Descript
 	}
 
 	if !b.ztocBuilder.CheckCompressionAlgorithm(compressionAlgo) {
-		return nil, fmt.Errorf("layer %s (%s) is compressed in an unsupported format. expect: [tar, gzip, unknown] but got %q: %w",
-			desc.Digest, desc.MediaType, compressionAlgo, errUnsupportedLayerFormat)
+		if compressionAlgo != compression.Gzip {
+			fmt.Printf("ztoc skipped - layer %s (%s) is compressed in an unsupported format. expect: [tar, gzip, unknown] but got %q",
+				desc.Digest, desc.MediaType, compressionAlgo)
+			return nil, errUnsupportedLayerFormat
+		}
```

Comment:
This may be a mistake in merge conflict resolution after the other half of uncompressed layer support landed while my PR was in draft. I will revisit this behavior and justify or correct it.